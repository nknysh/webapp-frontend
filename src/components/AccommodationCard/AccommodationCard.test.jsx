import React from 'react';

import AccomodationCard from './AccommodationCard.jsx';

const defaultProps = {
  id: 'Test ID',
  availableToHold: false,
  currencyCode: 'Â£',
  title: 'Test Subject',
  description: 'Some description',
  moreInformation: 'More information',
  amenities: ['Shower', 'Internet', 'Fridge'],
  size: 100,
  brochures: [
    {
      displayName: 'Beach House Brochure',
      url: 'http://www.africau.edu/images/default/sample.pdf?placeholderName=3b4add73-a827-4861-9064-ae3153d881df',
      uuid: '3b4add73-a827-4861-9064-ae3153d881df',
    },
    {
      displayName: 'Beach House Floor Plan',
      url: 'http://www.africau.edu/images/default/sample.pdf?placeholderName=3b4add73-a827-4861-9064-ae3153d881df',
      uuid: '3b4add73-a827-4861-9064-ae3153d881df',
    },
  ],
  totals: {
    oneOrMoreItemsOnRequest: false,
    total: '12345.00',
    totalBeforeDiscount: '23456.00',
    totalBeforeDiscountForPricedItems: '23456.00',
    totalBeforeDiscountForPricedItemsCents: 2345600,
    totalForPricedItems: '12345.00',
    totalForPricedItemsCents: 1234500,
  },
  occupancy: {
    standardOccupancy: 2,
    maximumPeople: 5,
    limits: [
      { maximum: 3, minimum: 1, name: 'default' },
      { maximum: 3, minimum: 1, name: 'Infant' },
      { maximum: 3, minimum: 1, name: 'Young Child' },
    ],
  },
  selectedCount: 99,
  appliedOffers: ['Test Offer 1', 'Test Offer 2'],
  imageUri: 'test-url',
  updateInProgress: false,
  // We should use spies here, but it's nigght on impossible to
  // select the NumberSelect buttons to trigger these actions.
  onRoomAdd: () => {},
  onRoomRemove: () => {},
};

describe('<AccomodationCard />', () => {
  describe('render', () => {
    it('matches snapshots (default props)', () => {
      const subject = shallow(<AccomodationCard {...defaultProps} />);
      expect(subject).toMatchSnapshot();
    });

    it('matches snapshots (Available to hold)', () => {
      const props = { ...defaultProps, availableToHold: true };
      const subject = shallow(<AccomodationCard {...props} />);
      expect(subject).toMatchSnapshot();
    });

    it('matches snapshots (Update in progress)', () => {
      const props = { ...defaultProps, updateInProgress: true };
      const subject = shallow(<AccomodationCard {...props} />);
      expect(subject).toMatchSnapshot();
    });
  });

  describe('More/Less Information', () => {
    // TODO: We can't test anything hook related until we update react to 16.8.5
  });
});
