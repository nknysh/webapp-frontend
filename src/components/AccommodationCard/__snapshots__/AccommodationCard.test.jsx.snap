// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccomodationCard /> render matches snapshots (Available to hold) 1`] = `
<div
  data-role="accomodation-cards"
>
  <p
    className="canHold"
  >
    Available to hold
  </p>
  <img
    data-role="ac-image"
    src="test-url"
  />
  <h1
    className="title"
    data-role="ac-title"
  >
    Test Subject
  </h1>
  <div
    className="innerWrapper"
  >
    <div
      className="info"
    >
      <ul
        className="occupancy"
      >
        <li
          data-role="ac-size"
        >
          Room size: 100 Square Meters
        </li>
        <li
          data-role="ac-standard-occupancy"
        >
          Standard occupancy - 2
        </li>
        <li
          data-role="ac-max-occupancy"
        >
          Max occupancy - 5
        </li>
      </ul>
      <ul
        className="minMax"
      >
        <li
          data-role="ac-min-max-item"
          key="default"
        >
          Adults
           - 
          Max
           
          3
          Min
           
          1
        </li>
        <li
          data-role="ac-min-max-item"
          key="Infant"
        >
          Infants
           - 
          Max
           
          3
          Min
           
          1
        </li>
        <li
          data-role="ac-min-max-item"
          key="Young Child"
        >
          Young Children
           - 
          Max
           
          3
          Min
           
          1
        </li>
      </ul>
      <ul
        className="actions"
      >
        <li
          data-role="ac-brochure"
          key="3b4add73-a827-4861-9064-ae3153d881df"
        >
          <Styled(Component)
            onClick={[Function]}
            title="Beach House Brochure"
          >
            + 
            Beach House Brochure
          </Styled(Component)>
        </li>
        <li
          data-role="ac-brochure"
          key="3b4add73-a827-4861-9064-ae3153d881df"
        >
          <Styled(Component)
            onClick={[Function]}
            title="Beach House Floor Plan"
          >
            + 
            Beach House Floor Plan
          </Styled(Component)>
        </li>
        <li>
          <Styled(Component)
            data-role="ac-more"
            onClick={[Function]}
            title="More information"
          >
            + 
            More information
          </Styled(Component)>
        </li>
      </ul>
    </div>
    <ul
      className="pricing"
    >
      <li>
        <label>
          Lodging Count:
           
          0
        </label>
        <br />
        <styled.button
          className="addLodgingButton"
          onClick={[Function]}
        >
          Add Lodging
        </styled.button>
      </li>
      <li>
        Total net cost
      </li>
      <li
        className="price"
      >
        £12,345.00
      </li>
      <li
        className="priceBeforeDiscount"
      >
        £23,456.00
      </li>
      <li
        data-role="ac-offer"
        key="Test Offer 1"
      >
        Test Offer 1
      </li>
      <li
        data-role="ac-offer"
        key="Test Offer 2"
      >
        Test Offer 2
      </li>
    </ul>
  </div>
</div>
`;

exports[`<AccomodationCard /> render matches snapshots (Update in progress) 1`] = `
<div
  data-role="accomodation-cards"
>
  <img
    data-role="ac-image"
    src="test-url"
  />
  <h1
    className="title"
    data-role="ac-title"
  >
    Test Subject
  </h1>
  <div
    className="innerWrapper"
  >
    <div
      className="info"
    >
      <ul
        className="occupancy"
      >
        <li
          data-role="ac-size"
        >
          Room size: 100 Square Meters
        </li>
        <li
          data-role="ac-standard-occupancy"
        >
          Standard occupancy - 2
        </li>
        <li
          data-role="ac-max-occupancy"
        >
          Max occupancy - 5
        </li>
      </ul>
      <ul
        className="minMax"
      >
        <li
          data-role="ac-min-max-item"
          key="default"
        >
          Adults
           - 
          Max
           
          3
          Min
           
          1
        </li>
        <li
          data-role="ac-min-max-item"
          key="Infant"
        >
          Infants
           - 
          Max
           
          3
          Min
           
          1
        </li>
        <li
          data-role="ac-min-max-item"
          key="Young Child"
        >
          Young Children
           - 
          Max
           
          3
          Min
           
          1
        </li>
      </ul>
      <ul
        className="actions"
      >
        <li
          data-role="ac-brochure"
          key="3b4add73-a827-4861-9064-ae3153d881df"
        >
          <Styled(Component)
            onClick={[Function]}
            title="Beach House Brochure"
          >
            + 
            Beach House Brochure
          </Styled(Component)>
        </li>
        <li
          data-role="ac-brochure"
          key="3b4add73-a827-4861-9064-ae3153d881df"
        >
          <Styled(Component)
            onClick={[Function]}
            title="Beach House Floor Plan"
          >
            + 
            Beach House Floor Plan
          </Styled(Component)>
        </li>
        <li>
          <Styled(Component)
            data-role="ac-more"
            onClick={[Function]}
            title="More information"
          >
            + 
            More information
          </Styled(Component)>
        </li>
      </ul>
    </div>
    <ul
      className="pricing"
    >
      <li>
        <label>
          Lodging Count:
           
          0
        </label>
        <br />
        <styled.button
          className="addLodgingButton"
          onClick={[Function]}
        >
          Add Lodging
        </styled.button>
      </li>
      <li>
        Total net cost
      </li>
      <li
        className="price"
      >
        £12,345.00
      </li>
      <li
        className="priceBeforeDiscount"
      >
        £23,456.00
      </li>
      <li
        data-role="ac-offer"
        key="Test Offer 1"
      >
        Test Offer 1
      </li>
      <li
        data-role="ac-offer"
        key="Test Offer 2"
      >
        Test Offer 2
      </li>
    </ul>
  </div>
</div>
`;

exports[`<AccomodationCard /> render matches snapshots (default props) 1`] = `
<div
  data-role="accomodation-cards"
>
  <img
    data-role="ac-image"
    src="test-url"
  />
  <h1
    className="title"
    data-role="ac-title"
  >
    Test Subject
  </h1>
  <div
    className="innerWrapper"
  >
    <div
      className="info"
    >
      <ul
        className="occupancy"
      >
        <li
          data-role="ac-size"
        >
          Room size: 100 Square Meters
        </li>
        <li
          data-role="ac-standard-occupancy"
        >
          Standard occupancy - 2
        </li>
        <li
          data-role="ac-max-occupancy"
        >
          Max occupancy - 5
        </li>
      </ul>
      <ul
        className="minMax"
      >
        <li
          data-role="ac-min-max-item"
          key="default"
        >
          Adults
           - 
          Max
           
          3
          Min
           
          1
        </li>
        <li
          data-role="ac-min-max-item"
          key="Infant"
        >
          Infants
           - 
          Max
           
          3
          Min
           
          1
        </li>
        <li
          data-role="ac-min-max-item"
          key="Young Child"
        >
          Young Children
           - 
          Max
           
          3
          Min
           
          1
        </li>
      </ul>
      <ul
        className="actions"
      >
        <li
          data-role="ac-brochure"
          key="3b4add73-a827-4861-9064-ae3153d881df"
        >
          <Styled(Component)
            onClick={[Function]}
            title="Beach House Brochure"
          >
            + 
            Beach House Brochure
          </Styled(Component)>
        </li>
        <li
          data-role="ac-brochure"
          key="3b4add73-a827-4861-9064-ae3153d881df"
        >
          <Styled(Component)
            onClick={[Function]}
            title="Beach House Floor Plan"
          >
            + 
            Beach House Floor Plan
          </Styled(Component)>
        </li>
        <li>
          <Styled(Component)
            data-role="ac-more"
            onClick={[Function]}
            title="More information"
          >
            + 
            More information
          </Styled(Component)>
        </li>
      </ul>
    </div>
    <ul
      className="pricing"
    >
      <li>
        <label>
          Lodging Count:
           
          0
        </label>
        <br />
        <styled.button
          className="addLodgingButton"
          onClick={[Function]}
        >
          Add Lodging
        </styled.button>
      </li>
      <li>
        Total net cost
      </li>
      <li
        className="price"
      >
        £12,345.00
      </li>
      <li
        className="priceBeforeDiscount"
      >
        £23,456.00
      </li>
      <li
        data-role="ac-offer"
        key="Test Offer 1"
      >
        Test Offer 1
      </li>
      <li
        data-role="ac-offer"
        key="Test Offer 2"
      >
        Test Offer 2
      </li>
    </ul>
  </div>
</div>
`;
